### PR TITLE
Issue #6933: Import auth provider plugins

### LINKED ISSUE #6933: Velero v1.12.0 has no auth provider found for name "oidc"
**What steps did you take and what happened:**

Velero server version: v1.12.0
Velero client version: v1.12.0
Velero client was not able to connect to Velero server when the auth provider is "oidc"

Ran velero --version command  and it threw an error **"An error occurred: no Auth Provider found for name "oidc""**

Velero v1.11.* are working fine when the auth provider is oidc but not working in v1.12.0

Upon debuggging I found out that pkg/client/auth_providers.go is present in v1.11.* but is removed in v1.12.0
v1.11.1 : https://github.com/vmware-tanzu/velero/blob/v1.11.1/pkg/client/auth_providers.go
v1.12.0: https://github.com/vmware-tanzu/velero/tree/v1.12.0/pkg/client

**What did you expect to happen:**

Velero client should be able to connect to Velero server and should display the client and server versions 

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by 0x113**
Thank you for contributing to Velero!

# Please add a summary of your change
Added imports for auth provider plugins, which were deleted by 180cc4e31d6b0ef7c62b9396cb834e81a9cef204

# Does your change fix a particular issue?

Fixes #6933 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by ywk253100** [comment]
@0x113 Could you sign-off the commit to pass the DCO checking?

**Comment by 0x113** [comment]
@ywk253100 I just signed it off. Thank you!

**Review by ywk253100 — DISMISSED** [review]

**Review by blackpiglet — DISMISSED** [review]

File: pkg/client/auth_providers.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+// Make sure we import the client-go auth provider plugins.
+
+import (
+	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
```
Comment (blackpiglet) [inline pkg/client/auth_providers.go:19]: According to the Revive testing data. It's better to move the comment line right above the import lines.

https://github.com/mgechev/revive/pull/501/files

```suggestion
package client

import (
	// Make sure we import the client-go auth provider plugins.
	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
		_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
)
```

File: pkg/client/auth_providers.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+// Make sure we import the client-go auth provider plugins.
```
Comment (blackpiglet) [inline pkg/client/auth_providers.go:16]: Move this line according to the following comment.

``` suggestion

```

File: pkg/client/auth_providers.go (hunk 1 above)
Comment (0x113) [reply pkg/client/auth_providers.go:18]: @blackpiglet, I moved the comment. Thank you for pointing that out!

**Review by blackpiglet — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**